resources/providers/azurelib/inventory/provider.go

```diff
@@ -224,3 +261,75 @@ func getString(data map[string]any, key string) string {
 	value, _ := data[key].(string)
 	return value
 }
+
+func transformDiagnosticSettingsClientListResponses(res []armmonitor.DiagnosticSettingsClientListResponse, subID, subName string) ([]AzureAsset, error) {
+	assets := make([]AzureAsset, 0, 8)
+
+	for i := range res {
+		for _, v := range res[i].Value {
+			if v == nil {
+				continue
+			}
+			a, err := transformDiagnosticSettingsResource(v, subID, subName)
+			if err != nil {
+				return nil, fmt.Errorf("error wile parsing azure asset model: %w", err)
+			}
+			assets = append(assets, a)
+		}
+	}
+
+	return assets, nil
+}
+
+func transformDiagnosticSettingsResource(v *armmonitor.DiagnosticSettingsResource, subID, subName string) (AzureAsset, error) {
+	properties, err := transformDiagnosticSettings(v.Properties)
+	if err != nil {
+		return AzureAsset{}, err
+	}
+
+	return AzureAsset{
+		Id:               strings.Dereference(v.ID),
+		Name:             strings.Dereference(v.Name),
+		Location:         "",
+		Properties:       properties,
+		ResourceGroup:    "",
+		SubscriptionId:   subID,
+		SubscriptionName: subName,
+		TenantId:         "",
+		Sku:              "",
+		Type:             strings.Dereference(v.Type),
+	}, nil
+}
+
+func transformDiagnosticSettings(d *armmonitor.DiagnosticSettings) (map[string]any, error) {
+	if d == nil {
+		return nil, nil
+	}
+
+	js, err := d.MarshalJSON()
+	if err != nil {
+		return nil, err
+	}
+
+	m := map[string]any{}
+	err = json.Unmarshal(js, &m)
+	if err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+func readPager[T any](ctx context.Context, pager *runtime.Pager[T]) ([]T, error) {
+	res := make([]T, 0, 8)
```

Comment:
```suggestion
	var res []T
```

let's go for simpler and consistent

Reply:
When we initialize the slice with nil and then append, we have some reallocation happening (check [this](https://go.dev/play/p/oQ0xdHsDt9k)).

Each capacity expansion (reallocation) doubles the previous capacity value. 

So its very common for example if you append 5 elements, to have 4 reallocation. 

So performance-wise is might be better to avoid that with a capacity (not fixed length) that might be large "enough" from the beginning. 

what do you think?

Reply:
I feel that when we are fetching information that potentially has to travel over a wire in the atlantic, optimizing for allocation of a couple of bytes is premature and makes reading the code slightly worse as the reader wonders "why 8". Still optional though.

Reply:
I will change it in this pr to nil and open a ticket/proposal to revisit our slices initialization in general, perhaps and we can discuss it there.